internal/tools/http/http.go

```diff
@@ -330,6 +331,12 @@ func (t Tool) Invoke(ctx context.Context, params tools.ParamValues) (any, error)
 		req.Header.Set(k, v)
 	}
 
+	if ua, err := util.UserAgentFromContext(ctx); err == nil {
+		req.Header.Set("User-Agent", ua)
+	} else if req.Header.Get("User-Agent") == "" {
+		req.Header.Set("User-Agent", "http-tool")
+	}
```

Comment:
We have to use the UserAgentFromContext for the user agent to be picked up in the pipeline. There is no need for the back up user agent.